### PR TITLE
Make cross-origin canvas drawing tests use a same-origin redirect.

### PR DESCRIPTION
**Pull request by jdm**
These tests either pass or maintain existing known failures in Firefox and Chrome, and expose the problem in our current implementation that #15887 is addressing.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16699)

### CODE REVIEW

File: tests/wpt/web-platform-tests/common/canvas-tests.js [hunk 1]
```diff
@@ -90,6 +90,7 @@ function addCrossOriginYellowImage()
     var img = new Image();
     img.id = "yellow.png";
     img.className = "resource";
-    img.src = get_host_info().HTTP_REMOTE_ORIGIN + "/images/yellow.png";
+    img.src = get_host_info().HTTP_ORIGIN + "/common/redirect.py?location=" +
+        get_host_info().HTTP_REMOTE_ORIGIN + "/images/yellow.png";
```
Comment (asajeffrey) [inline tests/wpt/web-platform-tests/common/canvas-tests.js]: Can we add a comment explaining why we're using a redirect? Should we test cross-origin images as well as same-origin redirects to cross-origin images?

**Comment by jdm** [comment]
Congratulations, your question turned a 5 line diff into a 700 line diff. I hope you're happy.

**Comment by jdm** [comment]
In short - each generated test for security properties now generates two variants: one that loads a cross-origin image directly, and one that loads a same-origin URL that redirects to a cross-origin one.

**Comment by asajeffrey** [comment]
LGTM, might be worth having @jgraham have a look, since this is a biggish change that will end up beig upstreamed to wpt at some point?

**Comment by jgraham** [comment]
lgtm too

---

Review status: 0 of 35 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/16699#-:-KkCFwPGm99zi0eAlzjP:btflqi5)*

**Pull request merged**